pkg/tsdb/azuremonitor/tokenprovider/token_provider_azure.go

```diff
@@ -18,27 +17,19 @@ var (
 )
 
 type azureAccessTokenProvider struct {
-	datasourceId      int64
-	datasourceVersion int
-	ctx               context.Context
-	cfg               *setting.Cfg
-	route             *plugins.AppPluginRoute
-	authParams        *plugins.JwtTokenAuth
+	cfg        *setting.Cfg
+	authParams *plugins.JwtTokenAuth
 }
 
-func newAzureAccessTokenProvider(ctx context.Context, cfg *setting.Cfg, ds *models.DataSource, pluginRoute *plugins.AppPluginRoute,
+func NewAzureAccessTokenProvider(cfg *setting.Cfg,
```

Comment:
You should still be able to pass context to this func. You can use the request context within the auth middleware.

Reply:
moved the tokenprovider initialization to the request context to be able to use its ctx